pkg/services/sqlstore/migrations/ualert/channel.go

```diff
@@ -26,11 +23,56 @@ type notificationChannel struct {
 }
 
 // channelsPerOrg maps notification channels per organisation
-type channelsPerOrg map[int64]map[interface{}]*notificationChannel
+type channelsPerOrg map[int64][]*notificationChannel
 
 // channelMap maps notification channels per organisation
 type defaultChannelsPerOrg map[int64][]*notificationChannel
 
+// Union type for receiver map that contains both uid -> receiver and ID -> receiver
+type uidOrID interface{}
+
+// Start channel and alert migrations. This creates Alertmanager configs as well as migrates receivers and creates the initial root-level route.
+// Returns per org maps containing: The alertmanager config, all migrated receivers, all migrated default receivers
+func (m *migration) setupAlertmanagerConfigs() (amConfigsPerOrg, map[int64]map[uidOrID]*PostableApiReceiver, map[int64]map[string]struct{}, error) {
+	// allChannels: channelUID -> channelConfig
+	allChannelsPerOrg, defaultChannelsPerOrg, err := m.getNotificationChannelMap()
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	amConfigPerOrg := make(amConfigsPerOrg, len(allChannelsPerOrg))
+	receiversPerOrg := make(map[int64]map[uidOrID]*PostableApiReceiver, len(allChannelsPerOrg))
+	defaultReceiversPerOrg := make(map[int64]map[string]struct{}, len(allChannelsPerOrg))
+	for orgID := range allChannelsPerOrg {
+		amConfig, ok := amConfigPerOrg[orgID]
+		if !ok {
```

Comment:
I had moved this over from the previous code, but I agree the happy-path here is no longer correct. The `!ok` will always be entered since this is where the `amConfig`s get created for the first time. I addressed this (and others) in https://github.com/grafana/grafana/pull/47291/commits/d4cae9db266ca7ab7d562660f00ce5238dad9f63